include/swift/AST/Expr.h

```diff
@@ -652,13 +652,27 @@ class TrailingCallArguments
     return *static_cast<const Derived *>(this);
   }
 
+  // Work around MSVC bug: can't infer llvm::trailing_objects_internal,
+  // even though we granted friend access to it.
   size_t numTrailingObjects(
+#if defined(_MSC_VER) && !defined(__clang__)
+      llvm::trailing_objects_internal::TrailingObjectsBase::OverloadToken<
```

Comment:
This seems really brittle. I still don't like it, but I don't have an alternative to play with right now.